spec/services/proofing/resolution/plugins/threatmetrix_plugin_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec.describe Proofing::Resolution::Plugins::ThreatMetrixPlugin do
+  let(:applicant_pii) { Idp::Constants::MOCK_IDV_APPLICANT_WITH_SSN }
+  let(:current_sp) { build(:service_provider) }
+  let(:proofer_result) do
+    instance_double(Proofing::DdpResult, success?: true, transaction_id: 'ddp-123')
+  end
+  let(:request_ip) { Faker::Internet.ip_v4_address }
+  let(:threatmetrix_session_id) { 'cool-session-id' }
+  let(:user_email) { Faker::Internet.email }
+
+  subject(:plugin) do
+    described_class.new
+  end
+
+  before do
+    allow(IdentityConfig.store).to receive(:lexisnexis_threatmetrix_mock_enabled).
+      and_return(false)
+    allow(plugin.proofer).to receive(:proof).and_return(proofer_result)
+  end
+
+  describe '#call' do
+    subject(:call) do
+      plugin.call(
+        applicant_pii:,
+        current_sp:,
+        request_ip:,
+        threatmetrix_session_id:,
+        timer: JobHelpers::Timer.new,
+        user_email:,
+      )
+    end
+
+    context 'ThreatMetrix is enabled' do
+      before do
+        allow(FeatureManagement).to receive(:proofing_device_profiling_collecting_enabled?).
+          and_return(true)
+      end
+
+      it 'calls the ThreatMetrix proofer' do
+        call
+        expect(plugin.proofer).to have_received(:proof)
+      end
+
+      it 'creates a ThreatMetrix associated cost' do
+        expect { call }.
+          to change {
+               SpCost.where(cost_type: :threatmetrix, issuer: current_sp.issuer).count
+             }.to eql(1)
```

Comment:
1. personal style preference for `do...end` for multi-line blocks which unfornately means adding parentheses to this bc of operator precedence
2. I think we want literal `to(1)` instead of `to(eql(1))`
```suggestion
          to(
            change do
              SpCost.where(cost_type: :threatmetrix, issuer: current_sp.issuer).count
           end.to(1),
         )
```

Reply:
I went with pulling out an `sp_cost_count` method, which gets you: 

```ruby
expect { call }.to change { sp_cost_count }.to(1)
```

All on one line